internal/client.go

```diff
@@ -407,19 +407,35 @@ type (
 
 	// ConnectionOptions is provided by SDK consumers to control optional connection params.
 	ConnectionOptions struct {
-		TLS                *tls.Config
+		// TLS configures a connection level security credentials.
+		TLS *tls.Config
+
+		// Authority specifies the value to be used as the :authority pseudo-header.
+		// This value only works only when TLS is nil.
```

Comment:
"Only works" might imply an error might happen?

Is it "only used when TLS is nil" ?